modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -215,6 +223,132 @@ impl ClientModuleInit for WalletClientInit {
             task_group: args.task_group().clone(),
         })
     }
+
+    /// Wallet recovery
+    ///
+    /// Query bitcoin rpc for history of addresses from last known used
+    /// addresses (or index 0 one) until MAX_GAP unused ones.
+    ///
+    /// Notably does not persist the progress of addresses being querried,
+    /// because it is not expected that it would take long enough to bother.
+    async fn recover(
+        &self,
+        args: &ClientModuleRecoverArgs<Self>,
+        _snapshot: Option<&<Self::Module as ClientModule>::Backup>,
+    ) -> anyhow::Result<()> {
+        let db = args.db().clone();
+        if db
+            .begin_transaction_nc()
+            .await
+            .get_value(&RecoveryFinalizedKey)
+            .await
+            .unwrap_or_default()
+        {
+            debug!(target: LOG_CLIENT_MODULE_WALLET, max_gap=MAX_GAP, "Recovery already complete before");
+            return Ok(());
+        }
+
+        let start_tweak_idx = match _snapshot {
+            Some(WalletModuleBackup::V0(backup)) => {
```

Comment:
I believe we need to override `supports_backup` to return true in the wallet client, similar to the mint client. Without that, the client will skip including the wallet module backup when it creates and sends an encrypted backup to the federation.

https://github.com/fedimint/fedimint/blob/9c9b1bda637222993bdbe5865e592cd6a37e26f2/fedimint-client/src/module/mod.rs#L626

Reply:
Oof.